src/components/popup/popup.tsx

```diff
@@ -24,22 +24,38 @@ export interface PopupCompProps extends PopupProps {
     anchor: Element | Point | null;
 }
 
+export interface PopupState {
+    style: React.CSSProperties;
+}
+
 @properties
-export class Popup extends React.Component<PopupCompProps> {
+export class Popup extends React.Component<PopupCompProps, PopupState> {
     public static defaultProps: Partial<PopupCompProps> = {
         open: false,
         anchorPosition: {vertical: 'bottom', horizontal: 'left'},
         popupPosition: {vertical: 'top', horizontal: 'left'},
         syncWidth: true
     };
 
+    public state: PopupState = {
+        style: {}
+    };
+
     private portal: Portal | null;
 
+    public componentWillMount() {
+        this.setState({style: this.createStyle(this.props)});
+    }
+
+    public componentWillReceiveProps(nextProps: PopupCompProps) {
+        this.setState({style: this.createStyle(nextProps)});
```

Comment:
If we are creating the style on WillMount why do we need this call as well? If I remove it no tests are failing so if we need it, the scenario is not tested.

Reply:
Imagine re-rendering the popup with different props. If we didn't update the style here, the popup would render with the old style. Later, `componentDidUpdate` would compare the style again, and force a re-render with updated style. But that would probably happen in the next animation frame, which can look bad.

To test that this is working correctly we would need to check that `componentDidUpdate()` didn't trigger a second re-render. Possible in theory, but isn't worth the trouble I guess.

Regarding componentWillMount, most of the time it sets style to an empty object, because the anchor ref is usually not provided at that point.

Reply:
Maybe re-rendering with different children? the style should be the same.